node/src/actors/chain_manager/handlers.rs

```diff
@@ -196,7 +196,7 @@ impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
                         self.transactions_pool.clear_commits();
 
                         // Mining
-                        if self.mining_enabled {
+                        if self.mining_enabled && self.sm_state == StateMachine::Synced {
```

Comment:
What happens if we don't check the current state?

Reply:
now you can mine data requests if you are in AlmostSynced or Synced, but only mining when you are Synced. The idea is that mining blocks and mining requests has the same requirement, to be in Synced state

Reply:
With this change we lose the following log lines when the node is not synced:

```text
1 data requests for this epoch
Reputation: 1, eligibility: 1, total: 1066769, active identities: 18260
```

Reply:
an affordable lost :P

Reply:
> now you can mine data requests if you are in AlmostSynced or Synced, but only mining blocks when you are Synced. The idea is that mining blocks and mining requests has the same requirement, to be in Synced state

I thought that the only reason why there is an `AlmostSynced` state is to avoid forks because nodes would mine a block and consolidate it instead of consolidating other candidates. There is no problem in creating commit transactions, right? The worst that could happen is that it is invalid, so they will only be broadcasted to the peers of this node, and those peers will detect it as invalid and stop broadcasting it.

Maybe we can even remove the `AlmostSynced` state?

Reply:
Remove 'AlmostSynced' is not trivial

Reply:
Anyway, I would prefer to not include this one line.